### PR TITLE
gh-101100: Fix Sphinx warnings about list methods

### PR DESCRIPTION
**Pull request by ghost**
The list methods have no entries in the document, so removed these references.

Fixed 7 sphinx warnings:
```
library/collections.rst:784: WARNING: py:meth reference target not found: list.append [ref.meth]
library/collections.rst:784: WARNING: py:meth reference target not found: list.append [ref.meth]
tutorial/datastructures.rst:143: WARNING: py:meth reference target not found: list.append [ref.meth]
tutorial/datastructures.rst:143: WARNING: py:meth reference target not found: list.pop [ref.meth]
tutorial/datastructures.rst:342: WARNING: py:meth reference target not found: list.pop [ref.meth]
tutorial/datastructures.rst:495: WARNING: py:meth reference target not found: list.append [ref.meth]
tutorial/datastructures.rst:495: WARNING: py:meth reference target not found: list.extend [ref.meth]
```

* Issue: gh-101100

----
📚 Documentation preview 📚: https://cpython-previews--127054.org.readthedocs.build/

### CODE REVIEW

**Review by vstinner — APPROVED** [review]
LGTM.

List methods are not documented in a way that we can refer to them: https://docs.python.org/dev/library/stdtypes.html#mutable-sequence-types

File: Doc/tutorial/datastructures.rst [hunk 1]
```diff
@@ -142,8 +142,8 @@ Using Lists as Stacks
 
 The list methods make it very easy to use a list as a stack, where the last
 element added is the first element retrieved ("last-in, first-out").  To add an
-item to the top of the stack, use :meth:`~list.append`.  To retrieve an item from the
-top of the stack, use :meth:`~list.pop` without an explicit index.  For example::
+item to the top of the stack, use :meth:`!~list.append`.  To retrieve an item from the
+top of the stack, use :meth:`!~list.pop` without an explicit index.  For example::
```
Comment (m-aciek) [inline Doc/tutorial/datastructures.rst:146]: This change apparently broke the rendering, we don't want to display the tilde character. The exclamation mark should probably replace the tilde?

Zrzut ekranu 2024-12-6 o 11 52 28

Comment (ghost) [reply Doc/tutorial/datastructures.rst:146]: @m-aciek Thank you for pointing this out. I'll fix it soon.

Comment (hugovk) [reply Doc/tutorial/datastructures.rst:146]: Thanks, it'd be good to detect this with the Sphinx Lint too. I reported it at:

https://github.com/sphinx-contrib/sphinx-lint/issues/126

**Pull request merged**